src/Umbraco.Web/Redirects/RedirectUrlManagementController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Net.Http;
+using System.Text;
+using System.Web;
+using System.Web.Http;
+using System.Xml;
+using Umbraco.Core.Configuration;
+using Umbraco.Web.WebApi;
+using File = System.IO.File;
+
+namespace Umbraco.Web.Redirects
+{
+    public class RedirectUrlManagementController : UmbracoAuthorizedApiController
+    {
+        //add paging
+        [HttpGet]
+        public RedirectUrlSearchResult SearchRedirectUrls(string searchTerm, int page = 0, int pageSize = 10)
+        {
+            var searchResult = new RedirectUrlSearchResult { UrlTrackerDisabled = UmbracoConfig.For.UmbracoSettings().WebRouting.DisableRedirectUrlTracking };
+            var redirectUrlService = Services.RedirectUrlService;
+            var resultCount = 0L;
+
+            var redirects = string.IsNullOrWhiteSpace(searchTerm) 
+                ? redirectUrlService.GetAllRedirectUrls(page, pageSize, out resultCount)
+                : redirectUrlService.SearchRedirectUrls(searchTerm, page, pageSize, out resultCount);
+
+            searchResult.SearchResults = redirects;
+            searchResult.TotalCount = resultCount;
+            searchResult.CurrentPage = page;
+            //hmm how many results 'could there be ?
+            searchResult.PageCount = ((int)resultCount + pageSize - 1) / pageSize;
+
+            searchResult.HasSearchResults = resultCount > 0;
+            searchResult.HasExactMatch = (resultCount == 1);
+            return searchResult;
+
+        }
+
+        [HttpGet]
+        public HttpResponseMessage GetPublishedUrl(int id)
+        {
+            var publishedUrl = "#";
+            if (id > 0)
+            {
+                publishedUrl = Umbraco.Url(id);
+            }
+
+            return new HttpResponseMessage { Content = new StringContent(publishedUrl, Encoding.UTF8, "text/html") };
+
+        }
+
+        [HttpPost]
+        public HttpResponseMessage DeleteRedirectUrl(int id)
+        {
+
+            var redirectUrlService = Services.RedirectUrlService;
+            // has the redirect already been deleted ?
+            //var redirectUrl = redirectUrlService.GetById(redirectUrl.Id);
+            //if (redirectUrl== null)
+            //{
+            //    return new HttpResponseMessage(System.Net.HttpStatusCode.NotFound);
+            //}
+            redirectUrlService.Delete(id);
+            return new HttpResponseMessage(System.Net.HttpStatusCode.OK);
```

Comment:
An easier way for this is to return type `IHttpActionResult` and then just do `return Ok();`